pylint/checkers/base/name_checker/checker.py

```diff
@@ -659,7 +661,12 @@ def _check_typevar(self, name: str, node: nodes.AssignName) -> None:
         if name_arg is None and args and isinstance(args[0], nodes.Const):
             name_arg = args[0].value
 
-        if variance == TypeVarVariance.double_variant:
+        if variance == TypeVarVariance.inferred:
```

Comment:
```suggestion
        if variance is TypeVarVariance.inferred:
```

Reply:
All other comparisons with `TypeVarVariance` also use the equals sign.
Let's change it when we implement https://github.com/pylint-dev/pylint/issues/5356, maybe?

Reply:
Fine!